Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/Themes/InteractableColorTheme.cs

```diff
@@ -58,20 +83,213 @@ public override void SetValue(InteractableThemeProperty property, int index, flo
         {
             Color color = Color.Lerp(property.StartValue.Color, property.Values[index].Color, percentage);
 
+            // check if a text object exists and set the color,
+            // if not then fall back to renderer based color setting.
+            if (SetColorValue != null)
+            {
+                SetColorValue(color);
+            }
+            else
+            {
+                if (SetTextMeshProColor(color))
+                {
+                    SetColorValue = SetTextMeshProColor;
+                    return;
+                }
+
+                if (SetTextMeshProUGUIColor(color))
+                {
+                    SetColorValue = SetTextMeshProUGUIColor;
+                    return;
+                }
+
+                if (SetTextMeshColor(color))
+                {
+                    SetColorValue = SetTextMeshColor;
+                    return;
+                }
+
+                if (SetTextColor(color))
+                {
+                    SetColorValue = SetTextColor;
+                    return;
+                }
+            }
+
+            base.SetValue(property, index, percentage);
+
+        }
+
+        /// <summary>
+        /// Get color on UI Text
+        /// </summary>
+        /// <param name="success"></param>
+        /// <returns></returns>
+        protected bool GetTextColor(out Color color)
```

Comment:
nit: might want to use the name TryGetTextColor and document what color gets set to if you return false, and what returning false means.